pkg/registry/storage/volumeattachment/strategy.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package volumeattachment
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/storage/names"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	"k8s.io/kubernetes/pkg/apis/storage"
+	"k8s.io/kubernetes/pkg/apis/storage/validation"
+)
+
+// volumeAttachmentStrategy implements behavior for VolumeAttachment objects
+type volumeAttachmentStrategy struct {
+	runtime.ObjectTyper
+	names.NameGenerator
+}
+
+// Strategy is the default logic that applies when creating and updating
+// VolumeAttachment objects via the REST API.
+var Strategy = volumeAttachmentStrategy{legacyscheme.Scheme, names.SimpleNameGenerator}
+
+func (volumeAttachmentStrategy) NamespaceScoped() bool {
+	return false
+}
+
+// ResetBeforeCreate clears the Status field which is not allowed to be set by end users on creation.
+func (volumeAttachmentStrategy) PrepareForCreate(ctx genericapirequest.Context, obj runtime.Object) {
+}
+
+func (volumeAttachmentStrategy) Validate(ctx genericapirequest.Context, obj runtime.Object) field.ErrorList {
+	volumeAttachment := obj.(*storage.VolumeAttachment)
+	return validation.ValidateVolumeAttachment(volumeAttachment)
+}
+
+// Canonicalize normalizes the object after validation.
+func (volumeAttachmentStrategy) Canonicalize(obj runtime.Object) {
+}
+
+func (volumeAttachmentStrategy) AllowCreateOnUpdate() bool {
+	return false
+}
+
+// PrepareForUpdate sets the Status fields which is not allowed to be set by an end user updating a PV
+func (volumeAttachmentStrategy) PrepareForUpdate(ctx genericapirequest.Context, obj, old runtime.Object) {
+}
+
+func (volumeAttachmentStrategy) ValidateUpdate(ctx genericapirequest.Context, obj, old runtime.Object) field.ErrorList {
+	errorList := validation.ValidateVolumeAttachment(obj.(*storage.VolumeAttachment))
+	return append(errorList, validation.ValidateVolumeAttachmentUpdate(obj.(*storage.VolumeAttachment), old.(*storage.VolumeAttachment))...)
+}
+
+func (volumeAttachmentStrategy) AllowUnconditionalUpdate() bool {
+	return true
```

Comment:
Do you really want this to be true?

It will defeat the optimistic concurrency built around resourceVersion. See https://github.com/kubernetes/kubernetes/blob/02b3928a921b2e0930bf67b48f71c9dcd1086b4f/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L574-L579

Reply:
Nope. Changed it to false.